### PR TITLE
Added options for video and website embedding to template

### PR DESCRIPTION
**Pull request by JaroCamphuijsen**
Added options for video and website embedding to template, current options are `image`, `video`, `website` and `wide markdown` if another property is added to the chapter heading, an error message is shown.

### CODE REVIEW

File: pages/_story.vue [hunk 1]
```diff
@@ -19,16 +19,20 @@
       </div>
     </div>
 
-    <!-- Chapter image and description -->
+    <!-- Chapter description -->
     <div v-for="(chapter, idx) in chapters" v-show="idx===currentChapter" :key="idx" class="flex flex-row-reverse justify-end gap-2 overflow-auto h-full">
-      <div class="p-4 w-1/3 bg-white rounded overflow-auto">
+      <div v-if="chapter.props.widemd" class="p-4 w-full bg-white rounded overflow-auto" >
```
Comment (bvreede) [inline pages/_story.vue:24]:
```suggestion
      <div v-if="chapter.props.widemd" class="p-4 w-full bg-white rounded overflow-auto">
```

File: pages/_story.vue [hunk 2]
```diff
@@ -19,16 +19,20 @@
...
+        <div class="content-center" style="width:60%; margin:auto">
+            <nuxt-content :document="chapter" class="prose mb-6 max-w-none" />
```
Comment (bvreede) [inline pages/_story.vue:26]:
```suggestion
          <nuxt-content :document="chapter" class="prose mb-6 max-w-none" />
```
github actions is complaining about this 😅

File: pages/_story.vue [hunk 3]
```diff
@@ -37,8 +41,18 @@
           </div>
         </div>
       </div>
-      <div v-else>
-        <p> No image found for this chapter. Does the chapter tag for this story have an image key? e.g. :::Chapter{headline="Name of my chapter" image="chapimg.png"} </p>
+      <!-- Video from youtube -->
+      <div v-else-if="chapter.props.video" class="w-2/3 bg-white rounded">
+        <iframe class="object-contain w-full h-full max-w-full max-h-full mx-auto" v-bind:src="'https://www.youtube.com/embed/' + chapter.props.video" title="YouTube video player" frameborder="0"></iframe>
```
Comment (bvreede) [inline pages/_story.vue:46]:
```suggestion
        <iframe class="object-contain w-full h-full max-w-full max-h-full mx-auto" :src="'https://www.youtube.com/embed/' + chapter.props.video" title="YouTube video player" frameborder="0" />
```

Again, Github Actions was complaining about this (I tested this and it still works; v-bind seems to be optional here)

File: pages/_story.vue [hunk 4]
```diff
@@ -37,8 +41,18 @@
...
+      </div>
+      <!-- Website  -->
+      <div v-else-if="chapter.props.website" class="w-2/3 bg-white rounded">
+        <iframe class="object-contain w-full h-full max-w-full max-h-full mx-auto" v-bind:src="chapter.props.website" title="Website" frameborder="0"></iframe>
```
Comment (bvreede) [inline pages/_story.vue:50]:
```suggestion
        <iframe class="object-contain w-full h-full max-w-full max-h-full mx-auto" :src="chapter.props.website" title="Website" frameborder="0" />
```

File: static/stories/multimedia.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+## Wide Markdown
+
+This type of chapter is used for all markdown. The small left text panel becomes screen wide and there is no media panel. It can be done by adding the `widemd` prop:
+
+```markdown
+:::Chapter{headline="Wide Markdown" widemd=1}
+```
+
+Something has to be added in the widemd prop, 1 is generally a safe value.
```
Comment (bvreede) [inline static/stories/multimedia.md:58]:
```suggestion
Something has to be added in the widemd prop, 1 is generally a safe value. Note that 0 will not turn it off; the only way to disable `widemd` is simply to remove the prop entirely.
```

**Review by bvreede — APPROVED** [review]
Faaaab! Some small suggestions and notes; also made some change suggestions because there is an action that complains about the HTML so I followed instructions :)

File: pages/_story.vue (hunk 1 above)
[the remark this answers is not in the document]
Comment (JaroCamphuijsen) [reply pages/_story.vue]: I agree, it's not optimal and still sort of first version. I was also thinking about this. Maybe we need to have both chapter type and media content properties like:
`:::Chapter{headline="Video" type=video link="www.youtube.com/watch?v=IuWVAv0AxXY"}`

**Comment by JaroCamphuijsen** [comment]
I merged the reveal.js changes into this one and now they can exist next to one another. There are various things that need improvement (for example the widemd key behavior, as @bvreede pointed out). I suggest to put these in new issues and PRs.

**Pull request merged**